### PR TITLE
[Storage] Stream transactions

### PR DESCRIPTION
**Pull request by wqfish**
Similar to streaming account states, this adds the ability to stream a range of transactions.

### CODE REVIEW

File: storage/libradb/src/lib.rs [hunk 1]
```diff
@@ -768,6 +768,43 @@ impl LibraDB {
...
+            type Item = Result<Transaction>;
+
+            fn next(&mut self) -> Option<Self::Item> {
+                if self.next_version >= self.end_version {
+                    return None;
+                }
+
+                let ret = self
+                    .store
+                    .get_transaction_opt(self.next_version)
```
Comment (msmouse) [inline storage/libradb/src/lib.rs]: it'll be more efficient to use a DB iter (than point get)?

Comment (wqfish) [reply storage/libradb/src/lib.rs]: I think that's a good point. I tried a bit and the only issue I'm seeing is to do that the code will roughly become the following and it seems nontrivial to make the lifetime work. Any ideas?

```
struct TxnIter<'a> {
    store: Arc<TransactionStore>,
    inner: SchemaIterator<'a, TransactionSchema>,
}
```

Comment (msmouse) [reply storage/libradb/src/lib.rs]: Yeah seems no way without unsafe code (one way of using unsafe code is through `owning_ref::OwningHandle`) at this point. 

How about we change the interface? like:

```
let backup_handler = libra_db.get_backup_handler()  // wraps around `Arc<DB>`
let iter = backup_handler.get_txn_iter()  // good until the backup_handler goes out of scope
// use the iter
```

File: storage/libradb/src/ledger_store/mod.rs [hunk 2]
```diff
@@ -322,6 +341,30 @@ impl HashReader for LedgerStore {
     }
 }
 
+pub struct TransactionInfoIter<'a> {
+    inner: SchemaIterator<'a, TransactionInfoSchema>,
+    end_version: Version,
+}
+
+impl<'a> TransactionInfoIter<'a> {
+    fn next_impl(&mut self) -> Result<Option<TransactionInfo>> {
```
Comment (msmouse) [inline storage/libradb/src/ledger_store/mod.rs]: ditto what I said on the `TransactionIter`

File: storage/libradb/src/transaction_store/mod.rs [hunk 3]
```diff
@@ -70,5 +86,27 @@ impl TransactionStore {
     }
 }
 
+pub struct TransactionIter<'a> {
+    inner: SchemaIterator<'a, TransactionSchema>,
+    end_version: Version,
+}
+
+impl<'a> TransactionIter<'a> {
+    fn next_impl(&mut self) -> Result<Option<Transaction>> {
```
[the remark this answers is not in the document]
Comment (wqfish) [reply storage/libradb/src/transaction_store/mod.rs]: I guess I can add a bit assertion in the iterator. I don't like changing the API to return version.

**Pull request closed without merging**